lens/database.ml

```diff
@@ -302,5 +336,78 @@ module Insert = struct
       (fmt_vals |> Format.pp_comma_list)
       v.values fmt_returning ()
 
+  let split v =
+    let { table; columns; values; returning } = v in
+    let f v =
+      let values = [ v ] in
+      { table; columns; values; returning }
+    in
+    List.map ~f values
+
+  let no_returning v = { v with returning = [] }
+
+  let exec_insert_returning_hack ~db data =
+    let last_id_fun =
+      let driver = db.driver_name () in
+      match driver with
+      | "mysql" -> "last_insert_id()"
+      | "sqlite3" -> "last_insert_rowid()"
+      | _ ->
+          let fn = "exec_insert_returning_hack" in
+          E.Unsupported_for_driver { driver; fn } |> E.raise
+    in
+    let getid = Format.asprintf "SELECT %s as id" last_id_fun in
+    let exec_insert v =
+      let ins = Format.asprintf "%a" (fmt ~db) v in
+      Debug.print ins;
+      Statistics.time_query (fun () -> db.execute ins);
+      Debug.print getid;
+      Statistics.time_query (fun () ->
+          db.execute_select getid ~field_types:[ ("id", Phrase_type.Serial) ])
+    in
+    no_returning data |> split |> List.map ~f:exec_insert |> List.concat
+
+  let exec_insert_returning ~db ~field_types data =
+    match db.driver_name () with
+    | "sqlite3"
+     |"mysql" ->
+        exec_insert_returning_hack ~db data
+    | _ ->
+        let cmd = Format.asprintf "%a" (fmt ~db) data in
+        Debug.print cmd;
+        Statistics.time_query (fun () -> db.execute_select ~field_types cmd)
+
   (* optionally add a returning statement if required *)
 end
+
+module Crud = struct
```

Comment:
I'd prefer a more descriptive name such as "Update" or "DatabaseUpdate"

Reply:
I will rename this to `Change`, as anything `Update` related is even more confusing. Admittedly `Crud` requires querying as well, so wasn't correct anyways.